crates/re_viewer/src/app.rs

```diff
@@ -1709,71 +1043,24 @@ fn file_saver_progress_ui(egui_ctx: &egui::Context, app: &mut App) {
     }
 }
 
-// TODO(emilk): support saving data on web
-#[cfg(not(target_arch = "wasm32"))]
-fn save_buttons_ui(ui: &mut egui::Ui, app: &mut App) {
-    let file_save_in_progress = app.promise_exists(FILE_SAVER_PROMISE);
-
-    let save_button = Command::Save.menu_button(ui.ctx());
-    let save_selection_button = Command::SaveSelection.menu_button(ui.ctx());
-
-    if file_save_in_progress {
-        ui.add_enabled_ui(false, |ui| {
-            ui.horizontal(|ui| {
-                ui.add(save_button);
-                ui.spinner();
-            });
-            ui.horizontal(|ui| {
-                ui.add(save_selection_button);
-                ui.spinner();
-            });
-        });
-    } else {
-        ui.add_enabled_ui(app.store_db_is_nonempty(), |ui| {
-            if ui
-                .add(save_button)
-                .on_hover_text("Save all data to a Rerun data file (.rrd)")
-                .clicked()
-            {
-                ui.close_menu();
-                app.pending_commands.push(Command::Save);
-            }
-
-            // We need to know the loop selection _before_ we can even display the
-            // button, as this will determine whether its grayed out or not!
-            // TODO(cmc): In practice the loop (green) selection is always there
-            // at the moment so...
-            let loop_selection = app.loop_selection();
-
-            if ui
-                .add_enabled(loop_selection.is_some(), save_selection_button)
-                .on_hover_text(
-                    "Save data for the current loop selection to a Rerun data file (.rrd)",
-                )
-                .clicked()
-            {
-                ui.close_menu();
-                app.pending_commands.push(Command::SaveSelection);
-            }
-        });
-    }
-}
-
 #[cfg(not(target_arch = "wasm32"))]
-fn open(app: &mut App) {
+fn open_rrd_dialog() -> Option<StoreHub> {
```

Comment:
I know why this is, but this signature feels weird. I still think `StoreHub` needs a different name. This is really something like a `StoreBundle` whereas `Hub` implies to me something more persistent and interface-centric.

Reply:
Yeah, perhaps you're right